pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -206,6 +206,13 @@ func (a *acrProvider) Provide() credentialprovider.DockerConfig {
 			cfg[url] = *cred
 		}
 	}
+
+	// add ACR anonymous repo support: use empty username and password for anonymous access
+	cfg["*.azurecr.*"] = credentialprovider.DockerConfigEntry{
```

Comment:
What's the order of trying the matching? Will it always first try the empty user name one, or always the SP one? 
Which is most likely usage here? I mean depending on the usage, you may want to optimize for the most common scenario.

Reply:
@yuwaMSFT2 good question, now it's this order: "*.azurecr.io", "*.azurecr.*", which means use SP first, and then empty username/password, SP first should be the better choice in a long period.